include/psa/crypto_struct.h

```diff
@@ -285,19 +285,43 @@ typedef uint16_t psa_key_attributes_flag_t;
 #define MBEDTLS_PSA_KA_MASK_DUAL_USE (          \
         0)
 
+/* Only used when the key id doesn't encode the owners id, to fill the reserved
+ * field in psa_core_key_attributes_t
+*/
+#define MBEDTLS_KEY_ATTRIBUTE_RESERVED_INIT (int32_t) 0
+
 typedef struct {
     psa_key_type_t MBEDTLS_PRIVATE(type);
     psa_key_bits_t MBEDTLS_PRIVATE(bits);
     psa_key_lifetime_t MBEDTLS_PRIVATE(lifetime);
     mbedtls_svc_key_id_t MBEDTLS_PRIVATE(id);
+/* This ensures that size of struct doesn't change size depending on setting
+ * MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
+ */
+#if !defined(MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER)
+    int32_t  MBEDTLS_PRIVATE(reserved);
+#endif /* !MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER */
     psa_key_policy_t MBEDTLS_PRIVATE(policy);
     psa_key_attributes_flag_t MBEDTLS_PRIVATE(flags);
 } psa_core_key_attributes_t;
+_Static_assert(sizeof(psa_core_key_attributes_t) == 32, "Unexpected struct size.");
 
+/* Changing MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER changes the size
+ * of psa_core_key_attributes_t, which can lead to incompatibilties.
```

Comment:
No it doesn't (though it changes to number of members)? Unless you're referring to the Mbed TLS side of things.

Reply:
Agree, this comment doesn't really make sense. Georgios copied it from some old patch